src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -186,6 +186,11 @@ private IEnumerable<TypeInferenceInfo> InferTypeInArrowExpressionClause(ArrowExp
                     return InferTypeInBaseMethodDeclaration(arrowClause.Parent as BaseMethodDeclarationSyntax);
                 }
 
+                if (arrowClause.IsParentKind(SyntaxKind.GetAccessorDeclaration))
+                {
+                    return InferTypeInGetAccessorDeclaration(arrowClause.Parent as AccessorDeclarationSyntax);
+                }
```

Comment:
nit: consider rewriting all three of these 'ifs' as a switch.  
nit: just use pattern matching for all three checks.

Reply:
I updated it to use pattern matching, thanks for the suggestion!